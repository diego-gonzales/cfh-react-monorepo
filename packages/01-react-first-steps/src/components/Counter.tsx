import { useState } from 'react';

const Counter = ({ value }: { value: number }) => {
  console.log('render');
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter((c) => c + 1);
  };

  const handleSubstract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => setCounter(value);

  return (
    <div>
      <h2>Counter</h2>
      <p data-testid="test-counter">{counter}</p>

      <button aria-label="btn-add" onClick={handleSubstract}>
        -
      </button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
      <button aria-label="btn-substract" onClick={handleAdd}>
        +
      </button>
    </div>
  );
};

export default Counter;
