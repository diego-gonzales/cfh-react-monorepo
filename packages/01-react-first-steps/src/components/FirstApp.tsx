interface Props {
  title: string;
  subtitle?: string;
  name?: string;
}

export const FirstApp: React.FC<Props> = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};
