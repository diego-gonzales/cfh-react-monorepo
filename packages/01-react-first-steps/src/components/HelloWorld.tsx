interface Titles {
  title: string;
  counter: number;
}

const HelloWorld = ({ title, counter }: Titles) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{counter}</p>
    </>
  );
};

export default HelloWorld;
