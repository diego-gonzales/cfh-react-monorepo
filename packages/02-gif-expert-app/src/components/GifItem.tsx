import { type GifShorterInfo } from '../types'

interface Props {
  gif: GifShorterInfo
}

export const GifItem = ({ gif }: Props) => {
  return (
    <li>
      <div className="card w-full bg-base-100 shadow-xl">
        <figure>
          <img src={gif.image} alt={gif.title} className="h-52 object-cover" />
        </figure>
        <div className="card-body">
          <small className="text-center">{gif.title}</small>
        </div>
      </div>
    </li>
  )
}
