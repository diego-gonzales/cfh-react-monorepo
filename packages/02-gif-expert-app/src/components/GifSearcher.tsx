import { useState } from 'react'

interface Props {
  onSearch: (searchTerm: string) => void
}

export const GifSearcher = ({ onSearch }: Props) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()

    const valueToSearch = inputValue.trim()

    if (valueToSearch.length === 0) return

    onSearch(valueToSearch)
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search gifs"
        className="input input-md input-bordered input-primary w-full"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}
