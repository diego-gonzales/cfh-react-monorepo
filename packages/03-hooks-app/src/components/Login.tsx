import { useEffect, useState } from 'react'
import { Input } from './Input'
import { Message } from './Message'

export const Login = () => {
  const [myForm, setMyForm] = useState({
    email: 'Diego',
    password: 'diego@email.com',
  })

  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setMyForm({
      ...myForm,
      [event.target.name]: event.target.value,
    })
  }

  useEffect(() => {
    // console.log('useEffect called')
  }, [])

  useEffect(() => {
    // console.log('form changed')
  }, [myForm])

  useEffect(() => {
    // console.log('email field changed')
  }, [myForm.email])

  useEffect(() => {
    // console.log('password field changed')
  }, [myForm.password])

  return (
    <>
      <form className="max-w-lg">
        <div className="mb-2">
          <Input
            name="email"
            placeholder="Email"
            type="email"
            value={myForm.email}
            onChange={handleInputChange}
          />
        </div>

        <div>
          <Input
            name="password"
            placeholder="Password"
            type="password"
            value={myForm.password}
            onChange={handleInputChange}
          />
        </div>
      </form>

      {myForm.email === 'Diego Eduardo' && <Message />}
    </>
  )
}
