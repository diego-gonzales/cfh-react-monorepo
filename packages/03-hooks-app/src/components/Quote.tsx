import { FC } from 'react'
import { Loader } from './Loader'
import { Post } from '../interfaces/post.interface'

interface Props {
  post: Post | null
  isLoading: boolean
}

export const Quote: FC<Props> = ({ post, isLoading }) => {
  const { title, body } = post || {}

  return post ? (
    <div className="relative items-center block max-w-sm p-6 bg-white border border-gray-100 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-800 dark:hover:bg-gray-700">
      <h5
        className={`mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white ${
          isLoading && 'opacity-20'
        }`}
      >
        {title}
      </h5>
      <p
        className={`font-normal text-gray-700 dark:text-gray-400 ${
          isLoading && 'opacity-20'
        }`}
      >
        {body}
      </p>

      {isLoading && <Loader />}
    </div>
  ) : (
    <div>There is no posts</div>
  )
}
