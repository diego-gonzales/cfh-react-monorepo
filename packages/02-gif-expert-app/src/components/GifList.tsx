import { type GifShorterInfo } from '../types'
import { GifItem } from './GifItem'

interface Props {
  gifs: GifShorterInfo[]
}
export const GifList = ({ gifs }: Props) => {
  return (
    <>
      <ul className="grid place-items-center gap-4 grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4">
        {gifs.map((gif) => (
          <GifItem key={gif.id} gif={gif} />
        ))}
      </ul>
    </>
  )
}
