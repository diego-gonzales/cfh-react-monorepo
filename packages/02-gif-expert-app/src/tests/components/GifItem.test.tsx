/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
import { render, screen } from '@testing-library/react'
import { GifItem } from '../../components/GifItem'
import { type GifShorterInfo } from '../../types'

describe('Test GifItem', () => {
  test('should match snapshot', () => {
    const gif: GifShorterInfo = {
      id: '1',
      title: 'Naruto',
      image: 'https://one-punch.com/naruto.jpg',
    }

    const { container } = render(<GifItem gif={gif} />)
    expect(container).toMatchSnapshot()
  })

  test('should show the passed props correctly', () => {
    const gif: GifShorterInfo = {
      id: '2',
      title: 'Gok√∫',
      image: 'https://one-punch.com/goku.jpg',
    }

    render(<GifItem gif={gif} />)

    expect(screen.getByText(gif.title)).toBeTruthy()

    const imgElement = screen.getByRole('img') as HTMLImageElement

    expect(imgElement.src).toBe(gif.image)
    expect(imgElement.alt).toBe(gif.title)
  })
})
