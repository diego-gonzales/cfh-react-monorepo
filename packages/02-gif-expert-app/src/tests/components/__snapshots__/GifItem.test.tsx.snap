// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem should match snapshot 1`] = `
<div>
  <li>
    <div
      class="card w-full bg-base-100 shadow-xl"
    >
      <figure>
        <img
          alt="Naruto"
          class="h-52 object-cover"
          src="https://one-punch.com/naruto.jpg"
        />
      </figure>
      <div
        class="card-body"
      >
        <small
          class="text-center"
        >
          Naruto
        </small>
      </div>
    </div>
  </li>
</div>
`;
